lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -159,300 +210,403 @@
         reuse_existing_ip_allowed=dict(required=False, type='bool',
                                        default=False),
         release_on_disassociation=dict(required=False, type='bool', default=False),
         allow_reassociation=dict(type='bool', default=False),
         wait_timeout=dict(default=300, type='int'),
-        private_ip_address=dict(required=False, default=None, type='str')
+        private_ip_address=dict(required=False, default=None, type='str'),
+        name=dict(required=False, default=None, type='str'),
+        purge_tags=dict(required=False, default=False, type='bool'),
+        tags=dict(required=False, default={}, type='dict')
     ))
 
-    module = AnsibleModule(
+    module = AnsibleAWSModule(
         argument_spec=argument_spec,
         supports_check_mode=True,
         required_together=[
             ['device_id', 'private_ip_address'],
         ],
     )
 
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
+    warnings = []
 
-    ec2 = ec2_connect(module)
+    ec2 = module.client('ec2')
 
     device_id = module.params.get('device_id')
     instance_id = module.params.get('instance_id')
     public_ip = module.params.get('public_ip')
     private_ip_address = module.params.get('private_ip_address')
     state = module.params.get('state')
-    in_vpc = module.params.get('in_vpc')
-    domain = 'vpc' if in_vpc else None
+    ec2_classic = module.params.get('ec2_classic')
     reuse_existing_ip_allowed = module.params.get('reuse_existing_ip_allowed')
     release_on_disassociation = module.params.get('release_on_disassociation')
     allow_reassociation = module.params.get('allow_reassociation')
+    invpc = module.params.get('in_vpc')
+    name = module.params.get('name')
+    tags = module.params.get('tags')
+    purge_tags = module.params.get('purge_tags')
+
+    if name:
+        tags["Name"] = name
+
+    if invpc:
+        warnings.append("Option in_vpc is no longer used.  Please use ec2_classic.")
+        ec2_classic = not invpc
 
     if instance_id:
-        warnings = ["instance_id is no longer used, please use device_id going forward"]
-        is_instance = True
+        warnings.append("Option instance_id is no longer used.  Please use device_id going forward")
         device_id = instance_id
-    else:
-        if device_id and device_id.startswith('i-'):
-            is_instance = True
-        elif device_id:
-            if device_id.startswith('eni-') and not in_vpc:
-                module.fail_json(msg="If you are specifying an ENI, in_vpc must be true")
-            is_instance = False
 
     try:
-        if device_id:
-            address = find_address(ec2, public_ip, device_id, isinstance=is_instance)
-        else:
-            address = find_address(ec2, public_ip, None)
+        domain = determine_domain(ec2, device_id, ec2_classic)
+        address = find_address(ec2, name, public_ip, device_id)
```

Comment:
If state: absent and the ip is not there, this raises botocore exception `InvalidAddress.NotFound`.

Reply:
I resolved this with my rebase and force push.